catt/controllers.py

```diff
@@ -41,23 +43,43 @@ def __init__(self, app_name, app_id, supported_device_types):
 
 
 def get_chromecasts():
-    devices, _browser = pychromecast.get_chromecasts()
+    devices, browser = pychromecast.get_chromecasts()
     devices.sort(key=lambda cc: cc.name)
-    return devices
+    return (devices, browser.zc)
 
 
-def get_chromecast(device_name):
-    devices = get_chromecasts()
+def get_ccinfos(devices=None, zconf=None):
+    if zconf is None:
+        devices, zconf = get_chromecasts()
+    return [get_ccinfo(d, zconf) for d in devices]
+
+
+def get_ccinfo(device, zconf=None):
+    if zconf is None:
+        ip, port = device.host, device.port
+    else:
+        service_name = (
+            "-".join(device.model_name.split()) + "-" + str(device.uuid).replace("-", "") + "." + GOOGLE_ZCONF
```

Comment:
As per [this](https://github.com/home-assistant-libs/pychromecast/issues/381#issuecomment-651945670) comment, the `service_name` should be available from the `Chromecast` object, but I'm struggling to find just that.

Reply:
I'll give it a shot, I'll reconfigure my network so I can do discovery and look for it.